src/odemis/driver/test/xt_client_test.py

```diff
@@ -340,12 +343,16 @@ def test_set_beam_shift(self):
         new_beam_shift_x = beam_shift_range[1][0] - 1e-6
         new_beam_shift_y = beam_shift_range[0][1] + 1e-6
         self.scanner.beamShift.value = (new_beam_shift_x, new_beam_shift_y)
-        self.assertAlmostEqual((new_beam_shift_x, new_beam_shift_y), self.scanner.beamShift.value)
+        current_shift = self.scanner.beamShift.value
+        self.assertAlmostEqual(new_beam_shift_x, current_shift[0])
+        self.assertAlmostEqual(new_beam_shift_y, current_shift[1])
```

Comment:
@pieleric I found a few cases in odemis where `assertTupleAlmostEqual` has been defined in the test cases as a function. Does it make sense to move that function (in a separate PR) to `odemis.util.test`?

Reply:
Yes. The only reason I haven't done it is because I'm too lazy, but the more we wait, the more the work it is to change! So go ahead :-)
I think that it should then be called `assert_tuple_almost_equal()`.

Reply:
I'm implementing this and realized the easiest way to do it is to use `numpy.testing.assert_allclose()`

```
def assert_tuple_almost_equal(first, second, places=None, msg=None, delta=None):
    """
    check two tuples are almost equal (value by value)
    """
    numpy.testing.assert_allclose(first, second, rtol=10**-places, atol=delta, err_msg=msg)
```

The old functions do this:

```
def assertTupleAlmostEqual(self, first, second, places=None, msg=None, delta=None):
    """
    check two tuples are almost equal (value by value)
    """
    for f, s in zip(first, second):
        self.assertAlmostEqual(f, s, places=places, msg=msg, delta=delta)
```
But in odemis.util.test I cannot use `self.assertAlmostEqual`, because that's part of the testing class. Should I use the implementation with `assert_allclose`, or everywhere where `assertTupleAlmostEqual` is used should I replace it with `numpy.testing.assert_allclose`?

Reply:
It's not exactly identical. Because numpy has this "automatic dimension folding" mechanism, which shouldn't happen here. Ex: `numpy.testing.assert_allclose([1,1], 1)`  => Fine

So, I'd suggest to use your first proposal as `assert_tuple_almost_equal()`, but also checks that the two tuples are the same length. Note that the old version didn't exactly do that, as it only compares the elements present in both tuples (that's the behaviour of `zip()`, but that wasn't on purpose, so let's not reproduce that.